carto-package.json

```diff
@@ -0,0 +1,17 @@
+{
+    "name": "postgis",
+    "current_version": {
+        "requires": {
+            "postgres": ">=10.1.1"
+        },
+        "_comment_on_works_with": "Added here versions w/ proper PARALLEL support",
+        "works_with": {
```

Comment:
I don't understand this section. Shouldn't this be in the other packages requiring this one? I mean, top - bottom, not bottom - up. I'd be madness keeping this up to date.

Even if the final decision it's doing it this way, it would be lacking Windshaft and Mapnik at least (both use Postgis at certain point).

Reply:
Adding @lekum to comment on this.

To my understanding, this is a "soft dependency". Kind of a "recommends" for choosing component versions.

Reply:
In this case "postgis" doesn't "works with" dataservices or crackshaft there is no postgis-dataservices combo required. But it is in the reverse way, cranckshaft requires (not works with) posgis. Not sure if you get me 🤔

Reply:
Seeing https://github.com/CartoDB/odbc_fdw/pull/77#discussion_r209938867 I see that we are understanding `works_with` differently. I understood that it's kind of a soft dependency: for example Builder requires certain Windshaft-cartodb or the sql-api versions when running, but it's not a build or even runtime requirement, so it's declared as  `works_with`.
OTOH, Postgis doesn't require crankshaft at all to run, so it shouldn't declare it but the other way around it's true, so it should be declared only on crankshaft carto-package.json.

Am I understanding this incorrectly? I see that here you are declaring all Carto packages Postgis has any relation with, which means that if any package declares `works_with` Postgis, then Postgis needs an equivalent entry here. That would be a huge pain for both Postgis and specially Postgresql since everything `works_with` it :sweat_smile:

Reply:
Yes @Algunenano that is.

Reply:
"Reverse soft dependency" is the working definition I used in my head.

Reply:
> "Reverse soft dependency" is the working definition I used in my head.

I'm using literally the opposite definition :smile: 

> Yes, @Algunenano that is. "works_with" is a thing that we are going to use rarely, but the places where it should be used are clear. For example, a CARTO feature introduced in builder requires a version of windshaft and sql-api (or a feature of windshaft requires a version of sql-api and builder) this is the case where "works_with" enters the game, is not a build or even runtime requirement but a feature in order to be fully enabled requires a set of versions.

So, this means that we should remove the `works_with` from this PR, right? And I guess it's the same for the other ones (odbc_fdw, crankshaft)